src/phoenix/server/api/types/Experiment.py

```diff
@@ -21,12 +21,26 @@
 
 @strawberry.type
 class Experiment(Node):
+    ordinality: Private[Optional[int]] = None
     id_attr: NodeID[int]
     description: Optional[str]
     metadata: JSON
     created_at: datetime
     updated_at: datetime
 
+    @strawberry.field(
+        description="Sequence number (1-based) of experiments belonging to the same dataset"
+    )  # type: ignore
+    async def sequence_number(
+        self,
+        info: Info[Context, None],
+    ) -> int:
+        if self.ordinality is None:
+            self.ordinality = await info.context.data_loaders.experiment_sequence_number.load(
+                self.id_attr
+            )
+        return cast(int, self.ordinality)
+
```

Comment:
This strikes me as a place we should probably return a `ValueError` from the dataloader rather than an optional type. If the dataloader returns `None`, it will cause a GraphQL error, won't it?

Reply:
yes, i can add it. it's just of odd that we would have a `Experiment` object that couldn't find itself (i.e. it shouldn't exist).

Reply:
You can do an assert not None inside the dataloader.

Reply:
> inside the dataloader
because the detaloader is decoupled, it could get invalid ids, so returning None there is ok